### ISSUE TITLE
Requirements

### ISSUE THREAD

**Issue by amark** [bugs181/TimeGraph#1]
First of all - great work, excited to have you doing this.

Secondarily, I'll be being pretty nitpicky cause I have very specific requirement/goals I need:

Docs API looks pretty cool / ideal looking.

Not using `state` may lead to bugs - it should be default, with option to replace.

Any data **not** in a timegraph (what you are calling timeplot) structure defeats the point/purpose of scale.

Unless I'm misunderstanding how you can achieve performance without it in that structure (please correct if wrong), this is why noted starting from `lib/time`'s basis is important to begin with.

`lib/time` timegraph supports resolution up to 1ms "buckets" of resolution, not 1min, this is intentional.

Twitter does 8K tweets/sec. 1min bucket would mean nearly Half Million records that have to be loaded, potentially 24MB which is untenable.

Timegraph's performance is **not** designed to **at most** achieve Twitter level scale - Twitter is only 1 site! Decentralized web will be all sites, people, IoT, machines combined! Assume **at least** there may be 10K ~ 100K records in 1 bucket/resolution.

I like `.time(start, end)` with chaining. I also think some API for "scrolling" by amount (.first(X number)) is good, do note that those "amounts" might also need to be a range like time start/end, because in a chatroom for  example, on load you probably want "last 20 chats" then as you scroll and viewport might want to do "last 10 ~ 30" , "last 20 ~ 30" so there is always a scrolling buffer. So the API would need to also be able to handle the idea of a start/end or first/last range dynamicaly update. So far I like your htinking more on this direction with chaining than the original timegraph.

Please look into what you call timeplotting as the more important thing though, at scale & scrolling.

Thank you! My ETA is still assuming this will take 1 month, just FYI so don't overpromise to yourself!!

**Comment by amark** [comment]
Anything you need from me?

**Comment by bugs181** [comment]
Nope. All is good. Just rolling out a performance lib for Mocha tests. Thank you though!

**Comment by bugs181** [comment]
Update on my end. Refactored the chaining API and I'm much more happy with it. Will start adding range code back piece by piece. This was a fairly large refactor.
